pkg/apis/coordination/validation/validation.go

```diff
@@ -50,3 +53,58 @@ func ValidateLeaseSpec(spec *coordination.LeaseSpec, fldPath *field.Path) field.
 	}
 	return allErrs
 }
+
+// ValidateLeaseCandidate validates a LeaseCandidate.
+func ValidateLeaseCandidate(lease *coordination.LeaseCandidate) field.ErrorList {
+	allErrs := validation.ValidateObjectMeta(&lease.ObjectMeta, true, ValidLeaseCandidateName, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateLeaseCandidateSpec(&lease.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+func ValidLeaseCandidateName(name string, prefix bool) []string {
+	return utilvalidation.IsConfigMapKey(name)
+}
+
+// ValidateLeaseCandidateUpdate validates an update of LeaseCandidate object.
+func ValidateLeaseCandidateUpdate(lease, oldLeaseCandidate *coordination.LeaseCandidate) field.ErrorList {
+	allErrs := validation.ValidateObjectMetaUpdate(&lease.ObjectMeta, &oldLeaseCandidate.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateLeaseCandidateSpec(&lease.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateLeaseCandidateSpec validates spec of LeaseCandidate.
+func ValidateLeaseCandidateSpec(spec *coordination.LeaseCandidateSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(spec.LeaseName) == 0 {
+		allErrs = append(allErrs, field.Required(field.NewPath("leaseName"), "leaseName is required"))
+	}
+	ev := semver.Version{}
+	if spec.EmulationVersion != "" {
+		var err error
+		ev, err = semver.Parse(spec.EmulationVersion)
+		if err != nil {
+			fld := fldPath.Child("emulationVersion")
+			allErrs = append(allErrs, field.Invalid(fld, spec.EmulationVersion, "Must be a valid semver"))
+		}
+	}
+	bv := semver.Version{}
+	if spec.BinaryVersion != "" {
+		var err error
+		bv, err = semver.Parse(spec.BinaryVersion)
+		if err != nil {
+			fld := fldPath.Child("binaryVersion")
+			allErrs = append(allErrs, field.Invalid(fld, spec.BinaryVersion, "Must be a valid semver"))
+		}
+	}
+	if !bv.EQ(ev) && bv.LT(ev) {
+		fld := fldPath.Child("binaryVersion")
+		allErrs = append(allErrs, field.Invalid(fld, spec.BinaryVersion, "Must be greater or equal to EmulationVersion"))
+	}
```

Comment:
Marking this as resolved, but in a different way:

- Both must be set if strategy is OldestEmulationVersion
- I don't really think we should care if they're set and strategy isn't set to OldestEmulationVersion? It could be useful for the example case like another candidate setting OldestEmulationNewestBinaryVersion superceding the current strategy. The fields would be used for that strategy.